### DISCUSSION TITLE
What magit hook is right to run forge-pull automatically for the status buffer?

### DISCUSSION THREAD

**Discussion by ag91** [magit/forge#464] [Support] [answered]
Hi,

forge is an amazing help in working with GitHub! I usually open Magit with `magit-status` and it would be great if I could get `forge` updates at that time by running `forge-pull` automatically. I tried a few hooks but apparently they get invoked too often:
what do you hook `forge-pull` to?

Thanks!

**Comment by tarsius** [comment, marked as answer]
I would recommend against doing this. It seems unnecessary; fetching explicitly is simple enough and too slow to do it all the time. And as you have noticed automatically doing it when you want it to happen but not doing it when you don't want it to happen isn't trivial.

If you want to do it anyway then:

1. Advice `magit-status-setup-buffer`.
2. Make sure you use a `magit-bury-buffer-function` that buries the buffer but does not kill it.
3. Replace you global binding for `magit-status` to use `magit-status-quick` instead.

**Reply by ag91** [reply]
Oh, I thought it was a good idea! Thanks, I will follow your recommendation. I will just bind it to a key in the magit-status keymap, so that I don't have to type the whole command :)

**Reply by tarsius** [reply]
Will that binding be shorter than the default `f n`? :grinning:

**Reply by ag91** [reply]
oh! I didn't know about that: thanks again!! I think `(use-pakage :after magit)` doesn't setup my keybindings in the magit-status so I never saw it: I shall fix that. Amazing!

**Reply by DivineDominion** [reply]
Thanks for the `f n` hint -- didn't realize the `f` menu was expanded!
